src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1012,11 +1012,15 @@ class Simplex_tree {
    *
    * Any insertion, deletion or change of filtration value invalidates this cache,
    * which can be cleared with clear_filtration().  */
-  void initialize_filtration() {
+  void initialize_filtration(bool ignore_infinite_values = true) {
```

Comment:
Shouldn't the default be `false`, for consistency with the current behavior?

Reply:
Yes indeed, the simplex iterator by filtration values will be affected by this. Furthermore the majority of the filtrations used have no infinite values in general (Rips, alpha, cubical,...), I also think setting default value to `false` is better.